search-service/src/main/kotlin/com/egm/stellio/search/listener/EntityListener.kt

```diff
@@ -28,63 +27,69 @@ class EntityListener(
-                }
-            }
-            EventType.UPDATE -> {
-                // TODO add missing checks:
-                //  - existence of temporal entity attribute
-                //  - needs optimization (lot of JSON-LD parsing, ...)
-                val rawParsedData = jacksonObjectMapper().readTree(entityEvent.payload!!)
-                val rawEntity = expandJsonLdEntity(entityEvent.updatedEntity!!)
-                val attributeName = rawParsedData.fieldNames().next()
-                val attributeValuesNode = rawParsedData[attributeName]
+            EventType.CREATE -> processCreateEvent(entityEvent, content)
+            EventType.UPDATE -> processUpdateEvent(entityEvent)
+            EventType.APPEND -> logger.warn("Append operation is not yet implemented")
+            EventType.DELETE -> logger.warn("Delete operation is not yet implemented")
+        }
+    }
 
-                if (!attributeValuesNode.has("observedAt")) {
-                    logger.info("Ignoring update event for $attributeName, it has no observedAt information")
-                    return
+    private fun processCreateEvent(entityEvent: EntityEvent, content: String) {
+        try {
+            temporalEntityAttributeService.createEntityTemporalReferences(entityEvent.payload!!)
+                .subscribe {
+                    logger.debug("Bootstrapped entity (records created: $it)")
                 }
+        } catch (e: Exception) {
+            logger.error("Received a non-parseable entity : $content", e)
+        }
+    }
 
-                val expandedAttributeName = expandJsonLdKey(attributeName, rawEntity.contexts)!!
-                val rawAttributeValue = attributeValuesNode["value"]
-                val parsedAttributeValue =
-                    if (rawAttributeValue.isNumber)
-                        Pair(null, valueToDoubleOrNull(rawAttributeValue.asDouble()))
-                    else
-                        Pair(valueToStringOrNull(rawAttributeValue.asText()), null)
-                val datasetId = attributeValuesNode["datasetId"]?.asText()
+    private fun processUpdateEvent(entityEvent: EntityEvent) {
+        val rawEntity = expandJsonLdEntity(entityEvent.updatedEntity!!)
+        val jsonLdAttributes = JsonLdUtils.expandJsonLdFragment(entityEvent.payload!!, rawEntity.contexts)
+        val ngsiLdProperty = parseToNgsiLdAttributes(jsonLdAttributes)[0] as NgsiLdProperty
+        val propertyName = ngsiLdProperty.name
+        val temporalPropertyInstanceFound = ngsiLdProperty.instances.find { it.isTemporalAttribute() }
+        var temporalPropertyInstance: NgsiLdPropertyInstance?
+        if (temporalPropertyInstanceFound.nonEmpty()) {
+            temporalPropertyInstance = temporalPropertyInstanceFound.orNull()
+        } else {
+            logger.info("Ignoring update event for $propertyName, it has no observedAt information")
+            return
+        }
+
+        val (value, measuredValue) =
```

Comment:
Maybe this should go to the if block as well

Reply:
since we use return when temporalPropertyInstance is not found , there 's no need to put it in a if or else branch